### PR TITLE
[Frontend]Product Component

### PR DESCRIPTION
**Pull request by brentmillen**
## Ticket/s

- [Product Component](https://trello.com/c/jrhuXtr1/20-product-component)

## Changes

- Create product component
- Applied styled-component
- Added default prop to mantine theme for button
- Used readOnly Rating component from mantine

## Screenshot

- image

## How to test?

- checkout to branch
  - changes temporarily placed in SampleView
- access in browser http://localhost:5173/

_note: will remove the component in SampleView before merge once approved_

### CODE REVIEW

File: packages/ui/product/Product.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+`;
+
+const CardWrapper = styled.div`
+  margin: 48px auto 0;
+  width: 300px;
+  height: 600px;
+  // display: flex; TO DO - add this when used for the catalog page
+`;
+
+const CardDetails = styled.div`
```
Comment (dandalandone) [inline packages/ui/product/Product.tsx]: Move all styled components to styles.ts

File: packages/ui/product/Star.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import React, { FC } from 'react';
+
+interface SvgProps {
+  fill: string;
+}
+
+const Star: FC<SvgProps> = ({ fill }) => {
```
Comment (dandalandone) [inline packages/ui/product/Star.tsx]: Star component should accept value

**Review by dandalandone — DISMISSED** [review]
See request changes

**Comment by dandalandone** [comment]
Also make sure that this card is extendable.

File: packages/ui/product/Product.tsx (hunk 1 above)
Comment (brentmillen) [reply packages/ui/product/Product.tsx]: done in [e335910](https://github.com/dandalandone/POC-EyCommerce/pull/36/commits/e335910cb23a7361817e14053f9300ccb2fc5d9b)

File: packages/ui/product/Star.tsx (hunk 2 above)
Comment (brentmillen) [reply packages/ui/product/Star.tsx]: added the prop to Rating and set to readOnly. Renamed to StarSVG

changes in [1bf879c](https://github.com/dandalandone/POC-EyCommerce/pull/36/commits/1bf879c0e00e42d36a2ca52da0cf408e1490207d)

File: packages/ui/package.json [hunk 3]
```diff
@@ -9,15 +9,15 @@
     "generate:component": "turbo gen react-component"
   },
   "devDependencies": {
-    "@types/react": "^18.2.0",
-    "@types/react-dom": "^18.2.0",
     "eslint": "^8.42.0",
     "eslint-config-custom": "workspace:*",
     "react": "^18.2.0",
```
Comment (DanDalanDone69) [inline packages/ui/package.json:14]: move this to dependencies also

File: packages/ui/product/Product.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        <CardDetails>
+          <CardComponent className="product-name">
+            <Title order={4} weight={500}>
+              Spicy Korean Chicken BBQ
+            </Title>
+          </CardComponent>
+          <CardComponent className="product-rating">
+            <Rating
+              emptySymbol={<StarSVG fill="#CCCCCC" />}
+              fullSymbol={<StarSVG fill="#FFD500" />}
```
Comment (DanDalanDone69) [inline packages/ui/product/Product.tsx]: Upon re-checking no need to make component for star, convert it to.

Disregard my previous comment. Thanks Bren~

``` tsx
import { ReactComponent as StarSVG} from './star.svg';

<div>
   <StarSVG />
 </div>
```

File: packages/ui/product/StarSVG.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React, { FC } from 'react';
+
+interface SvgProps {
+  fill: string;
+}
+
+const StarSVG: FC<SvgProps> = ({ fill }) => {
```
Comment (DanDalanDone69) [inline packages/ui/product/StarSVG.tsx]: remove this component

File: packages/ui/product/styles.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  width: 300px;
+  height: 600px;
+  // display: flex; TO DO - add this when used for the catalog page
+`;
+
+export const CardDetails = styled.div`
+  margin: 1.25rem 0.1rem;
+`;
+
+export const CardComponent = styled.div`
```
Comment (DanDalanDone69) [inline packages/ui/product/styles.ts]: rename to `StyledDetailsContainer` or `wrapper`

File: packages/ui/product/Product.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import { CardWrapper, CardDetails, CardComponent } from './styles';
+import StarSVG from './StarSVG';
+import { Title, Card, Image, Text, Button, Rating } from '@mantine/core';
+interface ProductProps {
```
Comment (DanDalanDone69) [inline packages/ui/product/Product.tsx]: Interface name should start with `I` 

ex: `IProductProps`

**Review by DanDalanDone69 — DISMISSED** [review]
Please recheck comments. thanks!

Dan~

File: packages/ui/package.json (hunk 3 above)
Comment (brentmillen) [reply packages/ui/package.json:14]: done in prev commit

File: packages/ui/product/Product.tsx (hunk 4 above)
Comment (brentmillen) [reply packages/ui/product/Product.tsx]: StarSVG removed

File: packages/ui/product/StarSVG.tsx (hunk 5 above)
Comment (brentmillen) [reply packages/ui/product/StarSVG.tsx]: done

File: packages/ui/product/styles.ts (hunk 6 above)
Comment (brentmillen) [reply packages/ui/product/styles.ts]: Components renamed

File: packages/ui/product/Product.tsx (hunk 7 above)
Comment (brentmillen) [reply packages/ui/product/Product.tsx]: Added prefix

**Review by markjoshuacarmen — APPROVED** [review]

**Review by sandrex-SGV — APPROVED** [review]
LGTM!

**Pull request merged**